### PR TITLE
Reimplement tests

### PR DESCRIPTION
**Pull request by mrcaseb**
This is a new implementation of tests to be able to test on CRAN and make other devs find breaking changes in revdepcheck

We now have 2 raw pbp files and a games file for these two games and test pbp parsing as well as other pbp related functions with them.

The test uses expect_snapshot_file and compares to a locally saved csv. I am not sure if this will fail on other runners, when model outputs differ in the 8th decimal digit. We will see how test work in gh actions.

### CODE REVIEW

**Comment by mrcaseb** [comment]
tests are failing as expected. Probably need to round all numeric values to a fixed number of significant digits

**Comment by mrcaseb** [comment]
It would require a lot of code changes to  test `calculate_player_stats()` without downloading any files. It loads the multiple laterals file, it calls nflreadr::load_players(), and it loads the Dakota model.

I think we don't test it and hope that breaking changes, esp. to dplyr, will take effect in the pbp parser

**Review by tanho63 — COMMENTED** [review]
sketching some thoughts where instead of passing in a directory we pass in the games dataframe

File: R/helper_add_game_data.R [hunk 1]
```diff
@@ -5,21 +5,22 @@
...
-      fetched <- curl::curl_fetch_memory(url)
-
-      if (fetched$status_code %in% c(404, 500)) {
-        warning(warn <- 1)
+      # we use dir to specify the directory of a locally stored games file
+      # for unit tests
+      if (is.null(dir)){
+        games <- nflreadr::load_schedules()
+      } else {
+        games <- readRDS(file.path(dir, paste0("games.rds")))
```
[the remark this answers is not in the document]
Comment (tanho63) [reply R/helper_add_game_data.R:18]: ...and we might be able to do the same thing for players in calculate_player_stats? idk. not sure if `better` or `worse` but I like the rigidity of testing the top level functions...

**Comment by mrcaseb** [comment]
I would like to merge this now. We have 60% code coverage and can add tests for stats functions later

**Pull request merged**